src/test/java/com/zaxxer/sansorm/transaction/TxTransactionManagerTest.java

```diff
@@ -0,0 +1,74 @@
+package com.zaxxer.sansorm.transaction;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.sansorm.QueryTest;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.zaxxer.sansorm.SansOrm;
+import com.zaxxer.sansorm.SqlClosure;
+import com.zaxxer.sansorm.SqlClosureElf;
+
+public class TxTransactionManagerTest {
+   @Before // not @BeforeClass to have fresh table in each test
+   public void setUp() throws IOException {
+      QueryTest.setUpDataSourceWithSimpleTx();
+      SqlClosureElf.executeUpdate("CREATE TABLE tx_test (string VARCHAR(128))");
+   }
+
+   @After // not @AfterClass to have fresh table in each test
+   public void tearDown() {
+      SqlClosureElf.executeUpdate("DROP TABLE tx_test");
+      SansOrm.deinitialize();
+   }
+
+   @Test
+   public void shouldSupportNestedCalls() {
+      final Set<String> insertedValues = SqlClosure.sqlExecute(c -> {
+         SqlClosureElf.executeUpdate(c, "INSERT INTO tx_test VALUES (?)", "1");
+
+         // here goes nested SqlClosure
+         SqlClosure.sqlExecute(cNested -> SqlClosureElf.executeUpdate(cNested, "INSERT INTO tx_test VALUES (?)", "2"));
+
+         return getStrings(c);
+      });
+      assertThat(insertedValues).containsExactlyInAnyOrder("1", "2");
+   }
+
+   @Test
+   public void shouldRollbackHighestTx() {
+      assertThatThrownBy(() -> SqlClosure.sqlExecute(c -> {
+         SqlClosureElf.executeUpdate(c, "INSERT INTO tx_test VALUES (?)", "3");
+
+         // here goes nested SqlClosure
+         SqlClosure.sqlExecute(cNested -> {
+            SqlClosureElf.executeUpdate(cNested, "INSERT INTO tx_test VALUES (?)", "4");
+            throw new RuntimeException("boom!");
+         });
+
+         return SqlClosureElf.executeUpdate(c, "INSERT INTO tx_test VALUES (?)", "5");
+      })).isInstanceOf(RuntimeException.class).hasMessage("boom!");
+
+      final Set<String> insertedValues = SqlClosure.sqlExecute(TxTransactionManagerTest::getStrings);
+      assertThat(insertedValues).isEmpty();
```

Comment:
this test failed twice: first time because of unhandled RuntimeException (which led to commit instead of rollback) and second time because of default autocommit